R/sampling.R

```diff
@@ -23,7 +23,7 @@ inverse_sample <- function(n_samples, cdf = NULL, pdf = NULL) {
   U <- stats::runif(n_samples, 0, cdf$y[length(cdf$y)])
   X <- vector(mode = "numeric", length = n_samples)
   for (i in seq_len(n_samples)) {
-    for (j in seq_len(cdf$x)) {
+    for (j in 1:length(cdf$x)) {
       if (cdf$y[j] > U[i]) {
```

Comment:
Ah, that's my fault. I'd just put down `seq_len()` since it worked in the prior line, but `cdf$x` is a vector or list probably, so `seq_along()` would be the right answer here.